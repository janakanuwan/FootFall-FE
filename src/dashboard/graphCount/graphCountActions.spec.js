import {
  changeGraphDateRange,
  changeGraphDisplayOption,
  changeGraphDisplayType,
  setGraphData,
} from "./graphCountActions";

describe('graphActions', () => {
  it('should create an action to change graph display type', () => {
    expect(changeGraphDisplayType('IN')).toMatchSnapshot();
  });

  it('should create an action to set graph data', () => {
    expect(setGraphData([{name: 'Hello', IN: 1, OUT: 1, PRESENCE: 0}])).toMatchSnapshot();
  });

  it('should create an action to change the graph date range', () => {
    expect(changeGraphDateRange({type: 'FROM', date: '2018-04-04'})).toMatchSnapshot();
  });

  it('should create an action to change the graph display option', () => {
    expect(changeGraphDisplayOption('day')).toMatchSnapshot();
  });
});
