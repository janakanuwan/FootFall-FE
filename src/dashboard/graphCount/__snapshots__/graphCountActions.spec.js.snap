// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphActions should create an action to change graph display type 1`] = `
Object {
  "payload": "IN",
  "type": "CHANGE_GRAPH_DISPLAY_TYPE",
}
`;

exports[`graphActions should create an action to change the graph date range 1`] = `
Object {
  "payload": Object {
    "date": "2018-04-04",
    "type": "FROM",
  },
  "type": "CHANGE_GRAPH_DATE_RANGE",
}
`;

exports[`graphActions should create an action to change the graph display option 1`] = `
Object {
  "payload": "day",
  "type": "CHANGE_GRAPH_DISPLAY_OPTION",
}
`;

exports[`graphActions should create an action to set graph data 1`] = `
Object {
  "payload": Array [
    Object {
      "IN": 1,
      "OUT": 1,
      "PRESENCE": 0,
      "name": "Hello",
    },
  ],
  "type": "SET_GRAPH_DATA",
}
`;
